File: andrews-zack/who-likes-it.js
// only 4 possible cases based on what they could give us
// using an if/else statement to concat the indexes onto the string
// annoyingly long but...oh well

function likes(names) {
  if(names.length === 0) {
    return "no one likes this";
  } else if(names.length === 1) {
    return `${names[0]} likes this`;
  } else if(names.length === 2) {
    return `${names[0]} and ${names[1]} like this`;
  } else if(names.length === 3) {
    return `${names[0]}, ${names[1]} and ${names[2]} like this`;
  } else {
    return `${names[0]}, ${names[1]} and ${names.length - 2} others like this`;
  }
}

File: andrews-zack/who-likes-it.py
def likes(n):
    match len(n):
        case 0:
            return "no one likes this"
        case 1:
            return f"{n[0]} likes this"
        case 2:
            return f"{n[0]} and {n[1]} like this"
        case 3:
            return f"{n[0]}, {n[1]} and {n[2]} like this"
        case _:
            return f"{n[0]}, {n[1]} and {len(n)-2} others like this"
    pass
